react/LocationSearch.tsx

```diff
@@ -1,50 +1,95 @@
 import React, { useState } from 'react'
-import { Input, IconSearch, IconClear } from 'vtex.styleguide'
 import { FormattedMessage } from 'react-intl'
+import { Input, IconSearch, IconClear } from 'vtex.styleguide'
+import {
+  Combobox,
+  ComboboxInput,
+  ComboboxOption,
+  ComboboxPopover,
+  ComboboxList,
+} from '@reach/combobox'
+import '@reach/combobox/styles.css'
 
+import { addresses as mockedAddresses } from './addresses'
 import styles from './LocationSearch.css'
 
-const LocationSearch: React.FC = () => {
-  const [inputValue, setInputValue] = useState<string>('')
+const MAX_DROPDOWN_ADDRESSES = 6
+
+interface LocationSearchProps {
+  onSelectAddress?: (selectedAddress: string) => void
+}
+
+const LocationSearch: React.FC<LocationSearchProps> = ({ onSelectAddress }) => {
+  const [searchTerm, setSearchTerm] = useState<string>('')
 
   const handleKeyDown = (event: React.KeyboardEvent) => {
     if (event.key !== 'Escape') {
       return
     }
-    setInputValue('')
+    setSearchTerm('')
+  }
+
+  const handleAddressSelection = (selectedAddress: string) => {
+    setSearchTerm(selectedAddress)
+    onSelectAddress?.(selectedAddress)
+  }
+
+  // This function will be replaced in the future, after integrating the
+  // component with GraphQL queries.
+  const getAddresses = () => {
```

Comment:
I know this is temporary, but this is an expensive function so you should memoize its return value. I think the way it was before was good, you just needed to add an `useMemo`.
```ts
const addresses = useMemo(() => getAddresses(searchTerm), [searchTerm])
```